docs/contributing/code-reviewing.md

```diff
@@ -1,231 +1,364 @@
+   discussions on the [development community server][czo] to see if the
+   additional context helps. If any part is still confusing, post a GitHub
+   comment or a message on the [development community server][czo] explaining
+   precisely what points you find confusing.
+
+2. Now that you're confident that you understand the issue, **does the PR
+   address all the points described in the issue**? If not, is it easy to tell
+   without reading the code which points are not addressed and why? Here is a
+   handful of good ways for the author to communicate why the issue as written
+   might not be fully solved by the PR:
+
+   - The issue explicitly notes that it's fine for some parts to be completed
+     separately, and the PR clearly indicates which parts are solved.
+   - After discussion of initial prototypes (in GitHub comments or on the
+     [development community server][czo]), it was decided to change some part of
+     the specification, and the PR notes this.
+   - The author explains why the PR is a better way to solve the issue than what
+     was described.
+   - The solution changed because of changes in the app since the issue was
+     written, and the author explains the adjustments that were made.
+
+**Think about the code:**
+
+1. Make sure the PR uses **clear function, argument, variable, and test names.**
+   Every new piece of Zulip code will be read many times by other developers, and
+   future developers will `grep` for relevant terms when researching a problem, so
+   it's important that variable names communicate clearly the purpose of each
+   piece of the codebase.
+
+1. Make sure the PR **avoids duplicated code.** Code duplication is a huge
+   source of bugs in large projects and makes the codebase difficult to
+   understand, so we avoid significant code duplication wherever possible.
+   Sometimes avoiding code duplication involves some refactoring of existing
+   code; if so, that should usually be done as its own series of commits (not
+   squashed into other changes or left as a thing to do later). That series of
+   commits can be in the same pull request as the feature that they support, and
+   we recommend ordering the history of commits so that the refactoring comes
+   _before_ the feature. That way, it's easy to merge the refactoring (and
+   minimize risk of merge conflicts) if there are still user experience issues
+   under discussion for the feature itself.
+
+1. **Good comments** help. It's often worth thinking about whether explanation
+   in a commit message or pull request discussion should be included in
+   a comment, `/docs`, or other documentation. But it's better yet if
+   verbose explanation isn't needed. We prefer writing code that is
+   readable without explanation over a heavily commented codebase using
+   lots of clever tricks.
+
+1. Make sure the PR follows Zulip's **coding style**. See the Zulip [code-style]
```

Comment:
We generally avoid using the `[code-style]` Markdown shorthand syntax since it's harder to grep.

Reply:
I guess this is pre-existing.